packages/pluggableWidgets/image-native/src/components/BackgroundImage.tsx

```diff
@@ -0,0 +1,66 @@
+import { createElement, ReactNode, FunctionComponent, useState, useCallback } from "react";
+import { ImageStyle, LayoutChangeEvent, View } from "react-native";
+import { extractStyles } from "@mendix/pluggable-widgets-tools";
+import { ResizeModeEnum } from "../../typings/ImageProps";
+import { CustomImageObjectProps, onLayoutSetDimensions } from "../utils/imageUtils";
+import { DimensionsType, ImageIconSVG } from "./ImageIconSVG";
+import { DefaultImageStyle } from "../ui/Styles.js";
+
+export interface BackgroundImageProps {
+    name?: string;
+    source: CustomImageObjectProps;
+    initialDimensions?: DimensionsType;
+    children: ReactNode;
+    resizeMode: ResizeModeEnum;
+    opacity: number;
+    styles: DefaultImageStyle;
+}
+
+export const BackgroundImage: FunctionComponent<BackgroundImageProps> = props => {
+    const [dimensions, setDimensions] = useState<DimensionsType>();
+    const { source, initialDimensions, children, opacity, styles, name } = props;
+    const [svgProps] = extractStyles(styles.image as ImageStyle, ["width", "height"]);
+    const wrapperDimensions = useCallback(
+        () =>
+            dimensions?.width && dimensions?.height
+                ? { width: dimensions?.width ?? svgProps?.width, height: dimensions?.height ?? svgProps?.height }
+                : { width: "100%", height: "100%" },
+        [dimensions, svgProps]
+    );
+
+    const onLayoutSetDimensionsCallback = useCallback(
+        ({ nativeEvent: { layout } }: LayoutChangeEvent) =>
+            onLayoutSetDimensions(layout.width, layout.height, setDimensions, initialDimensions),
+        [initialDimensions]
+    );
+
+    return (
+        <View
+            testID={`${name}$ImageBackgroundView`}
+            onLayout={!dimensions?.width || !dimensions?.height ? onLayoutSetDimensionsCallback : undefined}
+            style={[wrapperDimensions(), styles.container]}
```

Comment:
Do you think it would be simpler to instead build up `style` without the use of `useCallback`? 

```
const hasWidthAndHeight = dimensions?.width && dimensions?.height;
const style = {
    width: hasWidthAndHeight ? dimensions?.width ?? svgProps?.width  : "100%",
    height: hasWidthAndHeight ? dimensions?.height ?? svgProps?.height : "100%",
    ...styles.container
}
```

Reply:
I removed the useCallback and moved the styles into the component 👍🏼